shared/test/unit/mocks/mock_service.js

```diff
@@ -28,8 +48,52 @@ var MockService = {
   },
   query: function(state) {
     switch (state) {
+      case 'isUpgrade':
+        return this.mIsUpgrade;
+      case 'InputWindowManager.isOutOfProcessEnabled':
+        return this.mIsOutOfProcessEnabled;
+      case 'fetchCurrentOrientation':
+        return this.mCurrentOrientation;
+      case 'globalOrientation':
+        return this.mGlobalOrientation;
+      case 'defaultOrientation':
+        return this.mDefaultOrientation;
+      case 'isDefaultPortrait':
+        return this.mIsDefaultPortrait;
+      case 'AppWindowManager.getActiveApp':
+        return this.mActiveApp;
+      case 'getApp':
+        return this.mGetApp;
+      case 'keyboardHeight':
+        return this.mKeyboardHeight;
+      case 'keyboardEnabled':
+        return this.mKeyboardEnabled;
+      case 'slowTransition':
+        return this.mSlowTransition;
+      case 'HomescreenWindowManager.ready':
+        return this.mHomescreenReady;
+      case 'getHomescreen':
+        return this.mHomescreen;
+      case 'locked':
+        return this.locked;
+      case 'LayoutManager.width':
+        return this.mLayoutManager_width;
+      case 'LayoutManager.height':
+        return this.mLayoutManager_height;
+      case 'getHeightFor':
+      case 'LayoutManager.getHeightFor':
+        return this.mLayoutManager_height -
+          (arguments[2] ? 0 : this.mKeyboardHeight);
+      case 'SoftwareButtonMnager.height':
+        return this.mSoftwareButtonManager_height;
+      case 'SoftwareButtonMnager.width':
+        return this.mSoftwareButtonManager_width;
+      case 'InputWindowManager.height':
+        return this.mInputWindowManager_getHeight;
+      case 'getFtuOrigin':
+        return 'app://ftu.gaiamobile.org';
```

Comment:
I am sure this(mock things in MockService instead of mocking it in tests) was requested by you in certain review :)